pkg/engine/check_pod_task.go

```diff
@@ -209,24 +230,44 @@ func (task *CheckPodTask) verifyLabels(ctx context.Context, pod *v1.Pod) error {
 	return nil
 }
 
-func (task *CheckPodTask) verifyPod(ctx context.Context, podMap *utils.SyncMap, obj interface{}, errs chan error) {
+func (task *CheckPodTask) verifyPod(ctx context.Context, re []*regexp.Regexp, podMap *utils.SyncMap, count int, obj interface{}, errs chan error) {
 	pod, ok := obj.(*v1.Pod)
 	if !ok {
 		errs <- fmt.Errorf("%s: unexpected object type %T, expected *v1.Pod", task.ID(), obj)
 		return
 	}
 
-	if _, ok := podMap.Get(pod.Name); ok {
-		status := string(pod.Status.Phase)
-		task.log.V(4).Info("Informer event", "pod", pod.Name, "status", status)
-		if err := task.verifyLabels(ctx, pod); err != nil {
-			errs <- err
-			return
+	for _, r := range re {
+		if r.MatchString(pod.Name) {
+			task.log.V(4).Info("Matched pod", "name", pod.Name)
+			if _, ok := podMap.Get(pod.Name); ok {
+				return
+			}
+			status := string(pod.Status.Phase)
+			task.log.V(4).Info("Informer event", "pod", pod.Name, "status", status)
+			if status != task.Status {
+				return
+			}
+			if err := task.verifyLabels(ctx, pod); err != nil {
+				errs <- err
+				return
+			}
+			if sz := podMap.Set(pod.Name, true); sz == count {
+				task.log.Info("Accounted for all pods")
+				errs <- nil
+				return
+			}
 		}
-		if sz := podMap.Delete(pod.Name); sz == 0 {
-			task.log.Info("Accounted for all pods")
-			errs <- nil
-			return
+	}
+}
+
+func exp2regexp(expr []string) ([]*regexp.Regexp, error) {
```

Comment:
nit:

```suggestion
func (task *CheckPodTask) exp2regexp(expr []string) ([]*regexp.Regexp, error) {
```

or maybe put it a util pkg?

Reply:
thanks! Moved to utils, added unit test.